canonical-web-and-design/vanilla-framework
# Build responsive side navigation drawer

## Done

Adds responsive styles for side navigation.
See **Tasks** below for details on specific areas touched.

Fixes #2934

## QA

Run `./run` or [demo](https://vanilla-framework-canonical-web-and-design-pr-2977.run.demo.haus/docs/examples/templates/maas-docs-grid)

Check steps below on the example template page:

- load docs on large screen, sidenav should be in the grid
- resize to small screen, sidenav should hide
- toggle sidenav open - it should open
- toggle it back
- resize screen back to large - snap back to grid
- resize to small, open side nav, resize to large with open side nav
- make screen small (height and width) so sidenav needs to be scrollable, check if that works
- QA small screen on real phone


## Screenshots

<img width="544" alt="Screenshot 2020-04-15 at 15 02 39" src="https://user-images.githubusercontent.com/83575/79340159-32736a00-7f2a-11ea-8b64-359a1612d13b.png">




### Tasks

- [x] move colors to themes
  - what to do with buttons? (buttons don't have themes)
- [x] buttons (use p-button vs own pattern classes)
- [x] icons for < >
- [ ] expand/collapse aria
  - [x] added `aria-controls`
  - [ ] `aria-hidden`/`aria-expanded` turns out problematic because technically we would need to change it on resize
- [x] don't animate when resizing the screen
  - possible but requires a bit of JS to toggle additional class and listen to resize event (see commit https://github.com/canonical-web-and-design/vanilla-framework/pull/2977/commits/54a488f042fa6a24875f56ed7f9cf0f45ba0f43f)
- [x] when sidenav is expanded on small screen, when you resize to bigger screen sidenav goes to content, but overlay stays open
- [ ] overflow hidden on body when drawer is open (?)
   - this will need to react to resize as well to prevent situation when JS adds this to body but doesn't take it away on resize
- [x] close on overlay click
